eventio/var_int.pyx

```diff
@@ -0,0 +1,268 @@
+import numpy as np
+cimport numpy as np
+
+
+cpdef unsigned_varint(const unsigned char[:] data, unsigned long offset=0):
+    length = get_length_of_varint(data[0])
+    value = parse_varint(data[offset:offset + length])
+    return value, length
+
+
+def varint(const unsigned char[:] data, unsigned long offset=0):
+    value, length = unsigned_varint(data, offset)
+    if (value & 1) == 1:  # Negative number;
+        return -(value >> 1) - 1, length
+    return value >> 1, length
+
+
+cpdef get_length_of_varint(const unsigned char first_byte):
+    if (first_byte & 0x80) == 0:
+        return 1
+    if (first_byte & 0xc0) == 0x80:
+        return 2
+    if (first_byte & 0xe0) == 0xc0:
+        return 3
+    if (first_byte & 0xf0) == 0xe0:
+        return 4
+    if (first_byte & 0xf8) == 0xf0:
+        return 5
+    if (first_byte & 0xfc) == 0xf8:
+        return 6
+    if (first_byte & 0xfe) == 0xfc:
+        return 7
+    if (first_byte & 0xff) == 0xfe:
+        return 8
+    return 9
+
+
+cpdef parse_varint(const unsigned char[:] var_int_bytes):
+    length = var_int_bytes.shape[0]
```

Comment:
Would it help to define also the type of `length` here?

Reply:
I don't think so, because length comes from cython, it is part of its wrapper functionality for byteobjects

Reply:
So its type  is fixes and known to cython